### PR TITLE
Background removal

### PR DESCRIPTION
**Pull request by KamranZafar19**
1. - In the source folder, all the code and files are updated.
2. - In-line styling removed.
3. - Junk files removed from fonts folder and dropdown picker from components.
4. - All the things are done as per requirements.

### CODE REVIEW

File: src/components/backgroundTemplates/backgroundColors.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import {Text, View, Image, FlatList, TouchableOpacity, Modal} from 'react-native';
+import styles from '../../styles/index';
+import * as util from '../../utilities/index';
+import ColorPanel from 'react-native-color-panel';
+import * as component from '../../components';
+import LinearGradient from 'react-native-linear-gradient';
+import Ionicons from 'react-native-vector-icons/Ionicons';
+
+
+const DATA = [
```
Comment (gohar-chattha) [inline src/components/backgroundTemplates/backgroundColors.js:11]: Move constant data used in the app into constants file

File: src/components/backgroundTemplates/backgroundColors.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            backgroundColor: 'black'
+        },
+];
+export const BackgroundColors = ({ setBackground }) => {
+    const [modalVisible, setModalVisible] = useState(false);
+    const colorPicker = (color) => {
+        setBackground(color);
+
+    }
+  const Item = (item, i) => {
```
Comment (gohar-chattha) [inline src/components/backgroundTemplates/backgroundColors.js:31]: CamelCase not used

Comment (gohar-chattha) [reply src/components/backgroundTemplates/backgroundColors.js:31]: Its  should be item not Item

File: src/components/backgroundTemplates/backgroundColors.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                </TouchableOpacity>
+                <ColorPanel
+                style={styles.recentFlatlist}
+                fullColor={true}
+                brightnessLowerLimit={0}
+                onColorChange={colorPicker}
+            />
+            <View style={styles.colorPickerSelect}>
+                <component.Button title={"Select Color"}
+                    loginTextStyle = {styles.loginTextStyle}
```
Comment (gohar-chattha) [inline src/components/backgroundTemplates/backgroundColors.js:80]: Why was this prop used?
You already have a prop to pass style classes what was the need to create and send through a new prop?
Also name of your prop makes the component look like its for login only however its supposed to be generic

File: src/components/button/accessButton.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import Ionicons from 'react-native-vector-icons/Ionicons';
+
+export const AccessButton = ({title, icon, fadeButtonStyle, onPress, style}) => {
+    return (
+        <TouchableOpacity activeOpacity={0.5} style={[styles.AccessButton, style]} onPress={onPress}>
+            <View style={[styles.fadeWrapper, fadeButtonStyle]} />
+            <View style={styles.picBtnWrapper}>
+                <Text style={styles.picBtnText}>{title}</Text>
+            </View>
+            <View style={styles.IconWrapper}><Ionicons name={icon} size={25} color= {util.color.white} /></View>
```
Comment (gohar-chattha) [inline src/components/button/accessButton.js:14]: camelCase not used 
IconWrapper -> iconWrapper

File: src/components/button/backButton.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {Text, TouchableOpacity, View} from 'react-native';
+import styles from '../../styles/index';
+import * as util from '../../utilities';
+import Ionicons from 'react-native-vector-icons/Ionicons';
+
+export const Back = () => {
+    return (
+        <View style={styles.BackWrapper}>
```
Comment (gohar-chattha) [inline src/components/button/backButton.js:9]: camelCase not used

File: src/components/button/backButton.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+            <TouchableOpacity onPress={() => util.back()}>
+                <Ionicons name='chevron-back-outline' size={25} color= {util.color.grey} />
+            </TouchableOpacity>
+            <Text style={styles.BackText}>Back</Text>
```
Comment (gohar-chattha) [inline src/components/button/backButton.js:13]: camelCase not used

File: src/components/button/socialLogin.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { Image, TouchableOpacity } from "react-native";
+import styles from "../../styles";
+
+
+export const SocialLogin = ({onPress,icon}) => {
+
+  return (
+      <TouchableOpacity onPress={onPress}>
+          <Image source={icon} style={styles.SocialLogo} />
```
Comment (gohar-chattha) [inline src/components/button/socialLogin.js:10]: camelCase not used

File: src/components/button/toggleSwitch.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+import * as util from '../../utilities/index';
+
+
+export const ToggleSwitch = () => {
+  const [isEnabled, setIsEnabled] = useState(false);
+  const toggleSwitch = () => setIsEnabled(previousState => !previousState);
+
+  return (
+    <View style={styles.toggleWrapper}>
+    {isEnabled ? (<Text style={styles.ActiveText}>Save Password</Text>) : (<Text style={styles.InActiveText}>Save Password</Text>)}
```
Comment (gohar-chattha) [inline src/components/button/toggleSwitch.js:13]: camelCase not used

File: src/components/countryDropdown/countryDropdown.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+      if (id) {
+        setSelectedData(id)
+            setModalVisible(!modalVisible)
+          }
+        }
+        const Item = (item, index) => {
+      
+    return (
+        <View>
+          <View style={{flex: 1, alignItems: 'center',paddingVertical: util.WP(2)}}>
```
Comment (gohar-chattha) [inline src/components/countryDropdown/countryDropdown.js:22]: inline styling not acceptable

File: src/components/countryDropdown/countryDropdown.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+                            style={styles.DropdownIcon}
+                        />
+                    </View>
+                </View>
+            </View>
+            <Modal
+            animationType="slide"
+            transparent={true}
+            visible={modalVisible} >
+            <View style={{flex: 1, justifyContent:'flex-end', bottom: util.WP(26)}}>
```
Comment (gohar-chattha) [inline src/components/countryDropdown/countryDropdown.js:54]: inline styling

File: src/components/countryDropdown/countryDropdown.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+              <View style={{backgroundColor: util.color.white, flex: 1/3, elevation: 5,marginHorizontal: util.WP(5),borderRadius: util.WP(2),}}>
```
Comment (gohar-chattha) [inline src/components/countryDropdown/countryDropdown.js:55]: inline styling

File: src/components/headerLogo/headerLogo.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {Text, View, Image} from 'react-native';
+import styles from '../../styles/index';
+export const Header = () => {
+    return (
+        <View style={styles.Header}>
+            <View style={styles.LogoWrapper}>
```
Comment (gohar-chattha) [inline src/components/headerLogo/headerLogo.js:7]: came case not used

File: src/components/headerLogo/headerLogo.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+                {/* <Image */}
+                    {/* style={Styles.logoSize} */}
+                    {/* source={require('../../assets/logo.png')} */}
+                {/* // /> */}
+            </View>
+            <Text style={styles.HeaderText}>taskio</Text>
```
Comment (gohar-chattha) [inline src/components/headerLogo/headerLogo.js:13]: camel case

File: src/components/recentImageLayout/recentAddImageLayout.js [hunk 14]
```diff
@@ -0,0 +1,101 @@
+import React, {useEffect, useState} from 'react'
+import { View, Text, Image, FlatList, TouchableOpacity, Modal} from 'react-native'
+import styles from '../../styles/index';
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import * as util from '../../utilities';
+import * as component from '../../components';
+
+const bucket_name = 'current-app'
```
Comment (gohar-chattha) [inline src/components/recentImageLayout/recentAddImageLayout.js:8]: Whats bucket name current app doing in background removal app ?

File: src/components/recentImageLayout/recentAddImageLayout.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+const DATA = [
+        {
+            id: '01',
+            image: 'IMG_0011.JPG.jpg',
+        },
+        {
```
Comment (gohar-chattha) [inline src/components/recentImageLayout/recentAddImageLayout.js:14]: move to constants file

File: src/navigation/index.js [hunk 16]
```diff
@@ -1,44 +1,161 @@
...
 import {Provider as PaperProvider} from 'react-native-paper';
 import {navigationRef} from '../utilities/navigation';
-import Login from '../pages/login/login';
-import Homes from '../pages/home/home'
+import userSelection from '../pages/auth/userSelection';
+import Login from '../pages/auth/login';
+import SignUp from '../pages/auth/signup';
+import HomePage from '../pages/home/home';
+import Profile from '../pages/profile/profile';
+import AppGallery from '../pages/appGallery/appGallery';
```
Comment (gohar-chattha) [inline src/navigation/index.js:13]: camel case not used

File: src/navigation/index.js [hunk 17]
```diff
@@ -1,44 +1,161 @@
...
+import { connect } from "react-redux"
 
 const Auth = createNativeStackNavigator();
 const Home = createNativeStackNavigator();
 const Stack = createNativeStackNavigator();
+const Tab = createBottomTabNavigator(); 
 
+function Profiles() {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
```
Comment (gohar-chattha) [inline src/navigation/index.js:26]: inline styling

File: src/navigation/index.js [hunk 18]
```diff
@@ -1,44 +1,161 @@
...
+      <Text>Profile</Text>
+    </View>
+  );
+}
+
+function SettingsScreen() {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
```
Comment (gohar-chattha) [inline src/navigation/index.js:34]: inline styling

File: src/pages/auth/login.js [hunk 19]
```diff
@@ -1,47 +1,153 @@
...
+        setPasswordError(true);
+        setPasswordErrorText('password required');
+      }
+    }
+  };
+  useEffect (() => {
+    util.googleSignin();
+  }, []);
+
+  const onGoogleButtonPress = async () => {
```
Comment (gohar-chattha) [inline src/pages/auth/login.js:45]: use prettier extension and enable auto indent feature  on save of a file in your VS code.
Code is poorly indented

File: src/pages/auth/login.js [hunk 20]
```diff
@@ -1,47 +1,153 @@
...
+                />
+            </View>
+            <component.ToggleSwitch />
+            <View style={styles.buttonWrapper}>
+              <component.Button title={'Log in'}
+                loginTextStyle = {styles.loginTextStyle}
+                style = {styles.backgroundPrimary}
+                onPress={login}
+              />
+              <View style={styles.SocialContainer}>
```
Comment (gohar-chattha) [inline src/pages/auth/login.js:125]: camel case not used

File: src/pages/auth/login.js [hunk 21]
```diff
@@ -1,47 +1,153 @@
...
+                <View style={styles.SocialTextWrapper}>
```
Comment (gohar-chattha) [inline src/pages/auth/login.js:126]: camel case not used

File: src/pages/home/home.js [hunk 22]
```diff
@@ -0,0 +1,168 @@
...
+        formdata.append('userId', props.userId)
+        formdata.append('date', currentTime.toString())
+        props.bgRemove(formdata);
+        if (props.background_remove !== null) {
+          setModalVisible(true)
+        }
+      });
+  }
+
+  const gallaryPicture = () => {
```
Comment (gohar-chattha) [inline src/pages/home/home.js:50]: typo

File: src/store/actions/auth/index.js [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        }
+        });
+        dispatch (TASKS.hideLoader ());
+          } catch (e) {
+            console.log(e);
+    }
+  };
+}
+export const register = (email, password, ConfirmPassword, countryName, isLogin = false) => {
```
Comment (gohar-chattha) [inline src/store/actions/auth/index.js:35]: why ConfirmPassword is different from the rest 
camel case not used

File: src/store/actions/home/index.js [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+                    data: formdata,
+                };
+                dispatch (TASKS.showLoader ());
+                axios(config)
+                    .then((response) => {
+                    const background_remove = JSON.stringify(response.data.data.base64);
+                        dispatch(bgRemoval(background_remove));
+                       dispatch (TASKS.hideLoader ()); 
+                    })
+                    .catch((error) => {
```
Comment (gohar-chattha) [inline src/store/actions/home/index.js:39]: Why are there two catch blocks for one api call performed ?
Reasoning?

File: src/components/backgroundTemplates/backgroundColors.js (hunk 1 above)
Comment (KamranZafar19) [reply src/components/backgroundTemplates/backgroundColors.js]: committed

File: src/components/backgroundTemplates/backgroundColors.js (hunk 3 above)
Comment (KamranZafar19) [reply src/components/backgroundTemplates/backgroundColors.js]: committed

File: src/components/backgroundTemplates/backgroundColors.js (hunk 2 above)
Comment (KamranZafar19) [reply src/components/backgroundTemplates/backgroundColors.js]: committed

File: src/components/backgroundTemplates/backgroundColors.js (hunk 3 above)
Comment (KamranZafar19) [reply src/components/backgroundTemplates/backgroundColors.js]: <TouchableOpacity style={[styles.loginBtn, style]} onPress={onPress}>
            <Text style={[styles.buttonText, buttonTextStyle]}>{title}</Text>
</TouchableOpacity>
"style" prop is used for touchable view and "buttonTextStyle" prop is used for text.and now its prop name changed as for generic use.

File: src/components/button/accessButton.js (hunk 4 above)
Comment (KamranZafar19) [reply src/components/button/accessButton.js]: committed

File: src/components/button/backButton.js (hunk 5 above)
Comment (KamranZafar19) [reply src/components/button/backButton.js]: committed

File: src/components/button/backButton.js (hunk 6 above)
Comment (KamranZafar19) [reply src/components/button/backButton.js]: committed

File: src/components/recentImageLayout/recentAddImageLayout.js (hunk 14 above)
Comment (KamranZafar19) [reply src/components/recentImageLayout/recentAddImageLayout.js]: the bucket name is used in Url path if the bucket name is changed in the future then we change it here.
now bucket name moved in constant file and fixed it.
{ uri: 'https://' + util.bucketName +'.s3.amazonaws.com/' + item.image}

File: src/components/button/socialLogin.js (hunk 7 above)
Comment (KamranZafar19) [reply src/components/button/socialLogin.js]: committed

File: src/components/button/toggleSwitch.js (hunk 8 above)
Comment (KamranZafar19) [reply src/components/button/toggleSwitch.js]: committed

File: src/components/countryDropdown/countryDropdown.js (hunk 9 above)
Comment (KamranZafar19) [reply src/components/countryDropdown/countryDropdown.js]: committed

File: src/components/countryDropdown/countryDropdown.js (hunk 10 above)
Comment (KamranZafar19) [reply src/components/countryDropdown/countryDropdown.js]: committed

File: src/components/countryDropdown/countryDropdown.js (hunk 11 above)
Comment (KamranZafar19) [reply src/components/countryDropdown/countryDropdown.js]: committed

File: src/components/headerLogo/headerLogo.js (hunk 12 above)
Comment (KamranZafar19) [reply src/components/headerLogo/headerLogo.js]: committed

File: src/components/headerLogo/headerLogo.js (hunk 13 above)
Comment (KamranZafar19) [reply src/components/headerLogo/headerLogo.js]: committed

File: src/components/recentImageLayout/recentAddImageLayout.js (hunk 15 above)
Comment (KamranZafar19) [reply src/components/recentImageLayout/recentAddImageLayout.js]: committed

File: src/navigation/index.js (hunk 16 above)
Comment (KamranZafar19) [reply src/navigation/index.js]: committed

File: src/pages/home/home.js (hunk 22 above)
Comment (KamranZafar19) [reply src/pages/home/home.js:50]: camera and gallery type is used from utilities in this function

File: src/store/actions/home/index.js (hunk 24 above)
Comment (KamranZafar19) [reply src/store/actions/home/index.js:38]: inner catch is used for .then and outer catch is used for try.

File: src/navigation/index.js (hunk 17 above)
Comment (KamranZafar19) [reply src/navigation/index.js]: committed

File: src/store/actions/auth/index.js (hunk 23 above)
Comment (KamranZafar19) [reply src/store/actions/auth/index.js]: committed

File: src/pages/auth/login.js (hunk 21 above)
Comment (KamranZafar19) [reply src/pages/auth/login.js]: committed

File: src/pages/auth/login.js (hunk 20 above)
Comment (KamranZafar19) [reply src/pages/auth/login.js]: committed

File: src/pages/auth/login.js (hunk 19 above)
Comment (KamranZafar19) [reply src/pages/auth/login.js:45]: ok

File: src/navigation/index.js (hunk 18 above)
Comment (KamranZafar19) [reply src/navigation/index.js]: committed

File: src/store/actions/home/index.js (hunk 24 above)
Comment (KamranZafar19) [reply src/store/actions/home/index.js]: but now fixed with single catch block

**Pull request merged**